tests/testBootprint.py

```diff
@@ -0,0 +1,246 @@
+
+#
+# Copyright 2008-2017  AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+
+from __future__ import absolute_import, division, print_function
+import unittest
+import tempfile
+
+import lsst.utils.tests
+import lsst.afw.geom as afwGeom
+import lsst.afw.detection as afwDet
+import lsst.afw.table as afwTable
+
+
+class BootprintTestCase(unittest.TestCase):
+    def setUp(self):
+        self.spanRad = 4
+        self.regionRad = 10
+        self.spans = afwGeom.SpanSet.spanSetFromShape(self.spanRad,
+                                                      afwGeom.Stencil.BOX)
+        minPoint = afwGeom.Point2I(-self.regionRad, -self.regionRad)
+        maxPoint = afwGeom.Point2I(self.regionRad, self.regionRad)
+        self.region = afwGeom.Box2I(minPoint, maxPoint)
+        self.schema = afwDet.PeakTable.makeMinimalSchema()
+        # Run the the constructors test to ensure the Bootprints are setUp
+        self.testConstructors()
```

Comment:
It is run twice for the testConstructors test. All tests only run themselves, setup and tearDown. I could separate it out into a not test function, or have the code sucked into the setUp, but then I would not have a test that clearly tests the constructors for the benefit of future readers.